app/controllers/api/v1/decks_controller.rb

```diff
@@ -11,15 +11,16 @@ def index
   
   # GET /api/v1/decks/:id
   def show
+    authorize @deck
     deck_json = DeckSerializer.new(@deck).as_json
     render json: { deck: deck_json }, status: :ok
   end
   
   # POST /api/v1/decks
   def create
     deck = Deck.new(deck_params)
-    deck.user = current_user
     authorize deck
+    deck.user = current_user
```

Comment:
Please revert this change. Deck should have a user before you are calling `authorize`.